packages/core/RNSentryAndroidTester/app/src/androidTest/java/io/sentry/react/RNSentrySDKTest.kt

```diff
@@ -0,0 +1,127 @@
+package io.sentry.react
+
+import android.content.Context
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import io.sentry.ILogger
+import io.sentry.Sentry
+import io.sentry.Sentry.OptionsConfiguration
+import io.sentry.android.core.AndroidLogger
+import io.sentry.android.core.SentryAndroidOptions
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class RNSentrySDKTest {
```

Comment:
These integration tests look great. I like that they are close to what the user will execute.

Can you please add similar tests for the RN defaults and finals? Ideally, we check that the values are set/overwritten as expected both when reading options from the file and when the options are set directly in the callback.

Reply:
Sounds like a good idea. I'll iterate on this 👍

Reply:
I've added some basic tests for the defaults and finals with 2b7a319efc391f4c403e7482ccaedb728306861f and added `RNSentrySDK` tests with ced915d5593eae951f38779c9dfde22abcb58305 verifying the default/final configuration and the overriding behaviour. Note that a minor refactoring was needed d236c82480960c22e96deac2da4bfa8ac6c49588 since it was not possible to mock the Sentry SDK for Android testing.